consensus/core/src/context.rs

```diff
@@ -107,16 +107,29 @@ impl Context {
 /// Explicitly avoid to make `[Clock]` cloneable to ensure that a single instance is shared behind an `[Arc]`
 /// wherever is needed in order to make sure that consecutive calls to receive the system timestamp
 /// will remain monotonically increasing.
-pub(crate) struct Clock {
+pub struct Clock {
     initial_instant: Instant,
     initial_system_time: SystemTime,
+    // `clock_drift` should be used only for testing
+    clock_drift: BlockTimestampMs,
```

Comment:
you think its worthwhile to gate with `#[cfg(test)]`?

Reply:
Hhh unfortunately that's not possible. Initially I thought of creating a separate `TestClock` struct for this purpose, but then I either had to convert `Clock` to an interface or to an enum and use the test/non test version. If we find our selfs that we need to keep doing more special-handling cases for clock, then I am happy to do it.

In the meantime, I can put a guard check on the `timestamp_utc_ms` to ensure that the `clock_drift` is set only in test environments:

```
        cfg_if::cfg_if! {
            if !#[cfg(test)] {
                assert!(self.clock_drift == 0, "Clock drift should be zero in non testing environments");   
            }
        }
```

Reply:
Good idea. I've opted in instead making a check though on the `timestamp_utc_ms` . Let me know if you find this ok.

Reply:
That works! Just wanted some check to protect against accidental misuse.